tests/core/test_model.py

```diff
@@ -67,15 +81,32 @@ def test_llama_3_instruct_model_works(no_op_tracer: NoOpTracer) -> None:
     llama_3_model = Llama3InstructModel()
 
     prompt = llama_3_model.to_instruct_prompt(
-        "Who likes pizza?",
-        "Marc and Jessica had pizza together. However, Marc hated it. He only agreed to the date because Jessica likes pizza so much.",
+        INSTRUCTION,
+        INPUT,
     )
 
     complete_input = CompleteInput(prompt=prompt)
     output = llama_3_model.complete(complete_input, no_op_tracer)
     assert "Jessica" in output.completion
 
 
+class FakeRichPrompt:
+    items: Sequence[int] = [1]
+    ranges: Mapping[str, Sequence[PromptRange]] = {}
+
+
+def test_text_control_raises_error_for_non_text_prompt(
+    no_op_tracer: NoOpTracer,
+) -> None:
+    llama_3_model = Llama3InstructModel()
+
+    with (
+        patch.object(PromptTemplate, "to_rich_prompt", return_value=FakeRichPrompt()),
+        pytest.raises(ValueError, match="Text control only valid for text prompts."),
+    ):
```

Comment:
while our linter says these can be put together, and its right, I personally think its more read-able when they are not. Feel free to disagree tho

Reply:
Yes, makes it more readable. Had to disable ruff for this line tho, as it doesn't like nested with-statement. I think it's a price worth paying